nodes/core/core/lib/debug/debug-utils.js

```diff
@@ -369,7 +370,24 @@ RED.debug = (function() {
         })
         menuOptionMenu.show();
     }
+
+    var stack = [];
+    var busy = false;
     function handleDebugMessage(o) {
+        if (o) { stack.push(o); }
+        if (!busy && (stack.length > 0)) {
+            busy = true;
+            reallyHandleDebugMessage(stack.shift(), function() {
```

Comment:
`reallyHandleDebugMessage` completes synchronously - there's no need to pass in a callback function here. You can setup the `setTimeout` after `reallyHandleDebugMessage` returns.